test/e2e/spec/e2e_spec.rb

```diff
@@ -1651,15 +1720,141 @@ def get_policy_id_from_decode(tx_decoded_mint_or_burn)
                                                         mint_burn = nil,
                                                         validity_interval)
         expect(tx_constructed).to be_correct_and_respond 202
+        expected_fee = tx_constructed['fee']['quantity']
+
+        # Can be decoded
+        tx_decoded = SHARED.transactions.decode(@wid_sha, tx_constructed["transaction"])
+        expect(tx_decoded).to be_correct_and_respond 202
+
+        expect(tx_decoded['id'].size).to be 64
+        decoded_fee = tx_decoded['fee']['quantity']
+        expect(expected_fee).to eq decoded_fee
+        # inputs are ours
+        expect(tx_decoded['inputs'].to_s).to include 'address'
+        expect(tx_decoded['inputs'].to_s).to include 'amount'
+        expect(tx_decoded['outputs']).not_to eq []
+        expect(tx_decoded['script_validity']).to eq 'valid'
+        expect(tx_decoded['validity_interval']['invalid_before']).to eq validity_interval["invalid_before"]
+        expect(tx_decoded['validity_interval']['invalid_hereafter']).to eq validity_interval["invalid_hereafter"]
+        expect(tx_decoded['collateral']).to eq []
+        expect(tx_decoded['collateral_outputs']).to eq []
+        expect(tx_decoded['metadata']).to eq metadata
+        expect(tx_decoded['deposits_taken']).to eq []
+        expect(tx_decoded['deposits_returned']).to eq []
+        expect(tx_decoded['withdrawals']).to eq []
+        expect(tx_decoded['mint']).to eq ({"tokens"=>[]})
+        expect(tx_decoded['burn']).to eq ({"tokens"=>[]})
+        expect(tx_decoded['certificates']).to eq []
+      end
+
+      it "Delegation" do
+        balance = get_shared_balances(@wid_sha)
+        expected_deposit = get_key_deposit
+        puts "Expected deposit #{expected_deposit}"
+
+        # Pick up pool id to join
+        pools = SHELLEY.stake_pools
+        pool_id = pools.list({ stake: 1000 }).sample['id']
+
+        # Join pool
+        delegation = [{
+                        "join" => {
+                                    "pool" => pool_id,
+                                    "stake_key_index" => "0H"
+                                  }
+                      }]
+
+        tx_constructed = SHARED.transactions.construct(@wid_sha, payment = nil,
+                                                        withdrawal = nil,
+                                                        metadata = nil,
+                                                        delegation,
+                                                        mint_burn = nil,
+                                                        validity_interval = nil)
+        # Check fee and deposit on joining
+        tx_decoded = SHARED.transactions.decode(@wid_sha, tx_constructed["transaction"])
+        expect(tx_decoded).to be_correct_and_respond 202
+
+        # TODO: although you can construct and decode delegation, deposit_taken /deposit_returned are not shown atm
```

Comment:
delegation is left to receive separate treatment. we are going to support two modes here. I will obviously return to decodeTransaction when this work is delivered